crates/dojo-erc/src/erc721/erc721.cairo

```diff
@@ -0,0 +1,208 @@
+// TODO: proper safe_transfer_from
+//       token URI - wat do?
+
+#[contract]
+mod ERC721 {
+    use array::ArrayTrait;
+    use option::OptionTrait;
+    use starknet::contract_address;
+    use starknet::ContractAddress;
+    use starknet::get_caller_address;
+    use traits::Into;
+    use traits::TryInto;
+    use zeroable::Zeroable;
+
+    #[event]
+    fn Transfer(from: ContractAddress, to: ContractAddress, token_id: u256) {}
+
+    #[event]
+    fn Approval(owner: ContractAddress, approved: ContractAddress, token_id: u256) {}
+
+    #[event]
+    fn ApprovalForAll(owner: ContractAddress, operator: ContractAddress, approved: bool) {}
+
+    struct Storage {
+        world: ContractAddress,
+        name: felt252,
+        symbol: felt252
+        // token_uris: LegacyMap<u256, felt252>,
+        // balances: LegacyMap<ContractAddress, u256>,
+        // owners: LegacyMap<u256, ContractAddress>,
+        // token_approvals: LegacyMap<u256, ContractAddress>,
+        // operator_approvals: LegacyMap<(ContractAddress, ContractAddress), bool>,
+    }
+
+    //
+    // Constructor
+    //
+
+    #[constructor]
+    fn constructor(world_addr: ContractAddress, token_name: felt252, token_symbol: felt252) {
+        world::write(world_addr);
+        name::write(token_name);
+        symbol::write(token_symbol);
+    }
+
+    //
+    // External
+    //
+
+    #[external]
+    fn mint(to: ContractAddress) {
+        assert(to.is_non_zero(), 'minting to zero address');
+
+        let token_id = commands::uuid();
+        let mut calldata = ArrayTrait::new();
+        calldata.append(to);
+        calldata.append(token_id);
+
+        // TODO: will this work? if so, not the greatest...
+        let world_address = world::read();
+        commands::execute(ERC721Mint, calldata.span());
```

Comment:
currently `commands` are only supported in systems unfortunately. i'm working on adding support for inline macros to cairo which will allow us to easily support them elsewhere

for now, you need to use the low level api

Reply:
Ok, will do.